File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GroupBox/Programming/MouseClickedPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.GroupBox.Programming
{
    public partial class MouseClickedPage : UserControl
    {
        public MouseClickedPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles Click event of the GroupBox control.
        /// Shows message that indicates whether the event is fired.
        /// </summary>
        private void mobjGroupBox_Click(object sender, EventArgs e)
        {
            MouseEventArgs mea = e as MouseEventArgs;
            if (mea != null)
            {
                if (mea.Button == MouseButtons.Right)
                { //Check for right button click
                    this.mobjButtonClickedLabel.Text = "Right clicked";
                }
                else if (mea.Button == MouseButtons.Left)
                { //Check for Left button click
                    this.mobjButtonClickedLabel.Text = "Left clicked";
                }
            }
            else
            {
                this.mobjButtonClickedLabel.Text = "Mouse clicked!";
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GroupBox/Programming/MouseClickedPage.vb
Namespace CompanionKit.Controls.GroupBox.Programming

	Public Class MouseClickedPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        ''' <summary>
        ''' Handles Click event of the GroupBox control.
        ''' Shows message that indicates whether the event is fired.
        ''' </summary>
        Private Sub mobjGroupBox_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjGroupBox.Click
            Dim args As MouseEventArgs = TryCast(e, MouseEventArgs)
            If (Not args Is Nothing) Then
                If (args.Button = MouseButtons.Right) Then
                    Me.mobjButtonClickedLabel.Text = "Right clicked"
                ElseIf (args.Button = MouseButtons.Left) Then
                    Me.mobjButtonClickedLabel.Text = "Left clicked"
                End If
            Else
                Me.mobjButtonClickedLabel.Text = "Mouse clicked!"
            End If

        End Sub
	End Class

End Namespace
